app/frontend/Dockerfile

```
@@ -0,0 +1,23 @@
+# Use an official Node.js runtime as a parent image
+FROM node:20-slim
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Install pnpm
+RUN npm install -g pnpm
```

Hadolint (2.14.0)

[warning] Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`

(DL3016)